tests/e2e/attention/BUILD.bazel

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# End-to-end attention tests.
+
+load("//build_tools/bazel:iree_e2e_generated_runner_test.bzl", "iree_generated_e2e_runner_test")
+
+package(
+    features = ["layering_check"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+py_binary(
+    name = "generate_e2e_attention_tests",
+    srcs = ["generate_e2e_attention_tests.py"],
+)
+
+###########################################################################
+##
+## LLVMCPU backend
+##
+###########################################################################
+
+# Default CPU backend.
+[iree_generated_e2e_runner_test(
+    name = "e2e_attention_cpu_%s_%s_%s_%s" % (dtype, dtype, dtype, size),
```

Comment:
Have you tested on other backends? CPU is good to start, but we'll want tests running on all supported backends.

Reply:
As far as I know, we can only test it on the CPU backend. @Groverkss, you can correct me if I am wrong.

Reply:
Can you clarify "we can only test it" a bit? Test cases/suites showing failures are perhaps more valuable than test cases showing successes at this stage. 

That's a big thing I want to solve by moving away from the current Bazel/CMake model and putting test suites in https://github.com/iree-org/iree-test-suites. We need to be testing what _should_ work instead of only what _does_ work.

Reply:
@Groverkss what is your opinion?

Reply:
We know that attention on CPU backend is accurate. For hip, we are currently working on accuracy.

However, what Scott says makes sense. We should add support for both CPU and HIP backend in the tests and if HIP backend fails, we should XFAIL it and land it.

I'm not sure however if we want to add tests for both CPU and HIP in the same patch. I was hoping we can get the entire test suite running on CPU backend, and have a follow up patch that adds support for HIP, and XFAIL's if required. I'm fine with adding both backends in the same patch, if you think that's a better idea. However, personally, I would rather see this patch landed soon. If HIP works without any issues, I would land it, otherwise put it on a later patch (which I can do).

Reply:
I understand Scott's point; however, I also prefer to land this patch as a checkpoint and continue adding new backends as needed. In the meantime, I can also focus on moving the e2e test suites to the new place.

Reply:
> we should XFAIL it and land it

The current tests/e2e code has no concept of XFAIL. That part of why I want us to focus on https://github.com/iree-org/iree/issues/17868 and moving out out of iree-org/iree and into https://github.com/iree-org/iree-test-suites.

Much of the code in that directory should be moved out of tree and deleted, especially the matmul and conv tests, which is why I've been hesitant to add any more code there following the same patterns. I've been trying to learn through this PR what the tests fundamentally need and what code we can salvage.